### PR TITLE
Add knowledge checks in ruby/a-very-brief-intro-to-cs lesson

### LINKED ISSUE #22926: Add missing Knowledge Checks, Learning Outcomes, and Knowledge Check links to Ruby/A Very Brief Intro to CS
Related to #22467

This task is to go through the lesson referenced in the title of this issue and (as applicable):

- Add missing Learning Outcomes
- Add missing Knowledge Checks
- Add answer links in the Knowledge Checks

If the lesson already has Learning Outcomes or Knowledge Checks they do not need to be added to. Many lessons only need answer links added.

Knowledge check links should consist of an `<a>` tag which adds the `knowledge-check-link` class and links to the part of the lesson (or the external resource that is linked in the lesson) where the answer to the Knowledge Check can be found. Knowledge Check links should **not** point to external resources that weren't linked to as part of the lesson content or assignment. You can link to a specific line in the lesson like so:

```html
Lorem ipsum dolor sit amet, <span id='example-knowledge-check'>consectetur adipiscing elit</span>, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to the page where the span with this id lives -->
<a class='knowledge-check-link' href='#example-knowledge-check'>How do you return a value from a function?</a>
```
Or you can link to just a section of the lesson by just linking to the section heading and following [these](https://github.com/TheOdinProject/curriculum/wiki/Contributing-to-Wiki#anchor-linking) rules for determining how to format the link. Example:

```html
### Lorem ipsum dolor sit amet

sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to a heading on the page-->
<a class='knowledge-check-link' href='#lorem-ipsum-dolor-sit-amet'>How do you return a value from a function?</a>
```

### Acceptance Criteria

- [ ] Lesson referenced in the title has Learning Outcomes, Knowledge Checks, and Knowledge Check links
- [ ] Knowledge Check links all link to the proper place in the lesson, or the proper external resource that was linked to earlier in the lesson
- [ ] Knowledge Check styling should **not** change

Please do not start work on this unless you've been assigned to it. Comment on this issue and ping me @wise-king-sullyman if you would like to be assigned to it.

Additionally, we would like for contributors to only take on one of these issues at a time. Please only request assignment for (and begin work on) an issue after the Learning Outcome/Knowledge Check/Knowledge Check link issue you're currently assigned to has been approved and merged.

### PR DESCRIPTION
**Pull request by asadrjhk**
- [x] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [x] The PR title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 
#### 1.Describe the changes made and include why they are necessary or important:

I have added knowledge checks in the Ruby - A Very Brief Intro to CS lesson.

#### 2. If this PR is related to an open issue please reference it with the `#` operator and the issue number (and any [relevant keywords](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests)) below:

closes #22926

### CODE REVIEW

File: ruby_programming/computer_science/lesson_a_very_brief_intro_to_cs.md [hunk 1]
```diff
@@ -31,3 +31,11 @@ This section contains helpful links to other content. It isn't required, so cons
 * [Wikipedia on Computer Science](http://en.wikipedia.org/wiki/Computer_science)
 * [Wikipedia on Algorithms](http://en.wikipedia.org/wiki/Algorithm)
 * [Map of Computer Science](https://youtu.be/SzJ46YA_RaA)
+
+### Knowledge Check
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+* <a class="knowledge-check-link" href="http://ed.ted.com/lessons/your-brain-can-solve-algorithms-david-j-malan">How to think about Algorithms?</a>
```
Comment (wise-king-sullyman) [inline ruby_programming/computer_science/lesson_a_very_brief_intro_to_cs.md:38]: Can you rephrase this? I'm not really sure what the question is here.

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
Looks pretty good, one change I'd like to see.

**Review by wise-king-sullyman — COMMENTED** [review]
@asadrjhk do you know if we use this pattern of putting a code block in our knowledge checks anywhere else? I don't *think* we do, and if not I'll need to get up with the rest of the team to discuss if it's a direction we want to go in.

**Comment by asadrjhk** [comment]
@wise-king-sullyman oh i understand i didn't see that pattern till now in my progress, let me know if have to change this

**Comment by wise-king-sullyman** [comment]
@asadrjhk Yeah, we don't want to go in that direction at this time.

File: ruby_programming/computer_science/lesson_a_very_brief_intro_to_cs.md [hunk 2]
```diff
@@ -31,3 +31,11 @@ This section contains helpful links to other content. It isn't required, so cons
 * [Wikipedia on Computer Science](http://en.wikipedia.org/wiki/Computer_science)
 * [Wikipedia on Algorithms](http://en.wikipedia.org/wiki/Algorithm)
 * [Map of Computer Science](https://youtu.be/SzJ46YA_RaA)
+
+### Knowledge Check
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+* <a class="knowledge-check-link" href="http://ed.ted.com/lessons/your-brain-can-solve-algorithms-david-j-malan">Write an Algorithm to count the number of people in the room?</a>
```
Comment (wise-king-sullyman) [inline ruby_programming/computer_science/lesson_a_very_brief_intro_to_cs.md:38]: I don't think there actually is a good question to come from this resource, I think it would be best to just remove it.

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]

**Review by wise-king-sullyman — APPROVED** [review]
Perfect! Thanks for taking the time to make these changes :slightly_smiling_face:
